internal/session/repository_session.go

```diff
@@ -375,6 +375,34 @@ func (r *Repository) TerminateCompletedSessions(ctx context.Context) (int, error
 	return rowsAffected, nil
 }
 
+// TerminateSessionIfPossible is called on connection close and will attempt to close the connection's
+// session if the following conditions are met:
+//  * sessions that have exhausted their connection limit and all their connections are closed.
+//	* sessions that are expired and all their connections are closed.
+//	* sessions that are canceling and all their connections are closed
+func (r *Repository) TerminateSessionIfPossible(ctx context.Context, connectionId string, sessionVersion uint32) (int, error) {
```

Comment:
I think this can be unexported now since it should only be called by the domain service function